engine/pong.cpp

```diff
@@ -136,24 +139,29 @@ void Pong::LoadScene(Scene scene)
             constexpr float verticalWallX = 1280;
 
             auto topWall = std::make_unique<Wall>(horizontalWallWidth, horizontalWallHeight);
-            topWall->SetPosition(glm::vec3(0.0f, horizontalWallY, 0.0f));
+            topWall->InitalizeComponents();
```

Comment:
What is the advantage of having InitializeComponents externally here, instead of in the constructor (which will be naturally called on line 141).

With this being external it seems easy for someone else (or you) to mix it up and not call it in a new copy/paste

Reply:
Hmm well since it is a pure virtual it will still need to manually be added to the constructor right? I figured it would kind of be the same in that it could get missed when creating a new object. 
At least for `OnStart()` it needs to be called externally because things won't be ready properly unless the object is first positioned and added to the gameobjects list. That may not be the case now for `InitalizeComponents`, but I imagine it could be sometime in the future? 

Eventually I did want to clean up this part of the code so that you don't need to copy paste the whole.
`std::make_unique<>`
`InitalizeComponents()`
`GetComponent<Transform>()->mPosition =`
block, with a function to do each time.

Reply:
Cool. Put that down as a TODO item and we will re-discuss it there. I think figuring out how to do that would be valuable